### PR TITLE
V1.2 - use bunshi

### PR DESCRIPTION
**Pull request by loganvolkers**
## Description of the change

Replace the internals of `jotai-molecules`  to use `bunshi`. This makes the modules interoperable, and provide a backwards-compatible and gradual adoption path from  `jotai-molecules`  to `bunshi`.

Should deploy after #24 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.2.0] - 2023-04-20
```
Comment (00salmon) [inline CHANGELOG.md:10]:
```suggestion
## [1.2.0] - 2023-09-21
```

File: src/legacy/useMolecule.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Molecule } from "./molecule";
+import { MoleculeScopeOptions } from "./MoleculeScopeOptions";
+
+export function useMolecule<T>(
+  m: Molecule<T>,
+  options?: MoleculeScopeOptions
+): T {
+  // const scopes = useScopes(options);
+  // const store = useStore();
+  return {} as T;
```
Comment (00salmon) [inline src/legacy/useMolecule.tsx:10]: not sure if this is supposed to be returning something else

**Review by 00salmon — COMMENTED** [review]
just a couple minor things

File: src/legacy/useMolecule.tsx (hunk 2 above)
Comment (loganvolkers) [reply src/legacy/useMolecule.tsx:10]: It's only used for type tests, so nothing in `legacy` is run at runtime

**Review by 00salmon — APPROVED** [review]
looks good to me ✅

**Pull request merged**